clients/src/test/java/org/apache/kafka/clients/consumer/DeprecatedMethodsKafkaConsumerTest.java

```diff
@@ -0,0 +1,1736 @@
+import org.apache.kafka.common.record.TimestampType;
+import org.apache.kafka.common.requests.AbstractResponse;
+import org.apache.kafka.common.requests.FetchRequest;
+import org.apache.kafka.common.requests.FetchResponse;
+import org.apache.kafka.common.requests.FetchResponse.PartitionData;
+import org.apache.kafka.common.requests.FindCoordinatorResponse;
+import org.apache.kafka.common.requests.HeartbeatResponse;
+import org.apache.kafka.common.requests.IsolationLevel;
+import org.apache.kafka.common.requests.JoinGroupRequest;
+import org.apache.kafka.common.requests.JoinGroupResponse;
+import org.apache.kafka.common.requests.LeaveGroupResponse;
+import org.apache.kafka.common.requests.ListOffsetRequest;
+import org.apache.kafka.common.requests.ListOffsetResponse;
+import org.apache.kafka.common.requests.OffsetCommitRequest;
+import org.apache.kafka.common.requests.OffsetCommitResponse;
+import org.apache.kafka.common.requests.OffsetFetchResponse;
+import org.apache.kafka.common.requests.SyncGroupResponse;
+import org.apache.kafka.common.serialization.ByteArrayDeserializer;
+import org.apache.kafka.common.serialization.Deserializer;
+import org.apache.kafka.common.serialization.StringDeserializer;
+import org.apache.kafka.common.utils.LogContext;
+import org.apache.kafka.common.utils.MockTime;
+import org.apache.kafka.common.utils.Time;
+import org.apache.kafka.common.utils.Utils;
+import org.apache.kafka.test.MockConsumerInterceptor;
+import org.apache.kafka.test.MockMetricsReporter;
+import org.apache.kafka.test.TestUtils;
+import org.junit.Assert;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+
+import java.nio.ByteBuffer;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Properties;
+import java.util.Set;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.Future;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.concurrent.atomic.AtomicReference;
+import java.util.regex.Pattern;
+
+import static java.util.Collections.singleton;
+import static java.util.Collections.singletonList;
+import static java.util.Collections.singletonMap;
+import static org.apache.kafka.common.requests.FetchMetadata.INVALID_SESSION_ID;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertNotSame;
+import static org.junit.Assert.assertNull;
+import static org.junit.Assert.assertTrue;
+import static org.junit.Assert.fail;
+
+@SuppressWarnings({"deprecation", "SameParameterValue"})
+public class DeprecatedMethodsKafkaConsumerTest {
```

Comment:
We need to keep testing the methods we're deprecating, so I copied the whole `KafkaConsumerTest`. Then, I updated `KafkaConsumerTest` to use the new `poll(Duration)`. Once we get rid of the deprecated methods in a future version, we can just delete this test.

Reply:
There has to be a better way to do this. For example, maybe we can extend `KafkaConsumerTest`?

Reply:
I see where you're coming from, but I unironically think this is a great way to do this.

In general, we just want to make sure that we continue testing the same conditions we currently are. Capturing a snapshot of our current tests is a pretty sure way to do this.

I considered deleting everything that isn't directly relevant to the `poll` methods, but then again, @ConcurrencyPractitioner is going deprecate a bunch more methods, so I think he'll just have to add most of it back.

If we subclass `KafkaConsumerTest`, then you have the test harness separated from the tests, and when you modify `KafkaConsumerTest` in the future, you have to worry about also modifying these tests.

We could also just duplicate the relevant tests in `KafkaConsumerTest`, but that would make that file almost twice as long, therefore harder to read. Also, at some point, we'll want to remove the deprecated methods. When that happens, you can just delete this whole file, as opposed to having to go through `KafkaConsumerTest` and disentangle the deprecated-method and non-deprecated-method tests.